nautobot/core/utils/filtering.py

```diff
@@ -121,7 +121,15 @@ def get_filterset_parameter_form_field(model, parameter, filterset=None):
     elif isinstance(field, (MultiValueDecimalFilter, MultiValueFloatFilter)):
         form_field = forms.DecimalField()
     elif isinstance(field, NumberFilter):
-        form_field = forms.IntegerField()
+        # If "choices" are passed, then when 'exact' is used in an Advanced
+        # Filter, render a dropdown of choices instead of a free integer input
+        if field.lookup_expr == "exact" and getattr(field, "choices", None):
+            form_field = forms.MultipleChoiceField(
+                choices=field.choices,
+                widget=StaticSelect2Multiple,
+            )
```

Comment:
Not sure about this - what if I want to filter on `exact=12345` for whatever reason and 12345 isn't one of the choices?